NWNXLib/Assert.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <cstdio>
+
+namespace NWNXLib {
+
+namespace Assert {
+
+#if TAR_RELEASE
+    #define ASSERT(condition) (void)0
+    #define ASSERT_MSG(condition, format, ...) (void)0
+    #define ASSERT_FAIL() (void)0
+    #define ASSERT_FAIL_MSG(format, ...) (void)0
+#else
+    #define ASSERT(condition) \
+        if (!(condition)) ::NWNXLib::Assert::Fail((#condition), __FILE__, __LINE__, nullptr)
```

Comment:
You have an unbound `if` here. Consider this code:

    if (a > b)
        ASSERT(a > c);
    else
        printf("b <= a");

The `else` will be bound to the `if` from assert, and the print will happen even if `a>b`.

You can add `; else` to your definition which will get rid of it, but a nicer way to do this is:

    #define ASSERT(condition) \
      do { \
        /* actual code */ \
        } while (0)

(other instances of this)